core/authors/views.py

```diff
@@ -184,4 +189,130 @@ def check_is_following(self, request, pk):
         
         return Response({
             "isFollowingUser": follow.exists()
-        }, status=200)
\ No newline at end of file
+        }, status=200)
+
+    # All posts the currently auth'd user can see of pk
+    # /author/{AUTHOR_ID}/posts
+    @action(detail=True, url_path="posts")
+    def posts(self, request, pk=None):
+        page = int(request.query_params.get("page", 0)) + 1 # Must offset page by 1
+        if page < 1:
+            return Response("Page number must be positive", status=400)
+
+        # TODO: size should be limited?
+        size = int(request.query_params.get("size", 50))
+        if size < 0:
+            return Response("Size must be positive", status=400)
+
+        if not pk:
+            # TODO should it be a text response?
+            return Response("You must specify an author.",status=400)
+
+        # Only return public posts if the user isn't authenticated
+        if request.user.is_anonymous:
+            posts = Posts.objects.all().filter(author=pk, visibility__in=["PUBLIC"])
+        else:
+            requestingAuthor = request.user.author.id # Should be guaranteed because we checked above
```

Comment:
2 things:
I could be completely wrong about this but I don't think the User model has an author attribute. The Author model does have a user attribute though

You should call get_author_url(authorId) and set requestingAuthor to that instead, because Follows store author URLs (this is needed for integration with external servers).
import get_author_url

Reply:
The field is `OneToOne` so it is available in both the `user` and `author`